common4j/src/main/com/microsoft/identity/common/java/controllers/BaseController.java

```diff
@@ -152,6 +155,99 @@ public abstract AuthorizationResult deviceCodeFlowAuthRequest(final DeviceCodeFl
+        final String methodName = ":acquireToken";
+
+        Logger.verbose(
+                TAG + methodName,
+                "Acquiring token..."
+        );
+
+        final AcquireTokenResult acquireTokenResult = new AcquireTokenResult();
+
+        //00) Validate MSAL Parameters
+        parameters.validate();
+
+        // Add default scopes
+        final Set<String> mergedScopes = addDefaultScopes(parameters);
+
+        final RopcTokenCommandParameters parametersWithScopes = parameters
+                .toBuilder()
+                .scopes(mergedScopes)
+                .build();
+
+        logParameters(TAG, parametersWithScopes);
+
+        //0) Get known authority result
+        parametersWithScopes.getPlatformComponents()
+                .getPlatformUtil()
+                .throwIfNetworkNotAvailable(parametersWithScopes.isPowerOptCheckEnabled());
+
+        Authority.KnownAuthorityResult authorityResult = Authority.getKnownAuthorityResult(parametersWithScopes.getAuthority());
+
+        //0.1 If not known throw resulting exception
+        if (!authorityResult.getKnown()) {
+            Logger.error(TAG + methodName, "Authority is not known.", authorityResult.getClientException());
+            throw authorityResult.getClientException();
+        }
+
+        // Build up params for Strategy construction
+        final OAuth2StrategyParameters strategyParameters = OAuth2StrategyParameters.builder()
+                .platformComponents(parameters.getPlatformComponents())
+                .build();
+
+        //1) Get oAuth2Strategy for Authority Type
+        @SuppressWarnings(WarningType.rawtype_warning) final OAuth2Strategy oAuth2Strategy = parametersWithScopes
+                .getAuthority()
+                .createOAuth2Strategy(strategyParameters);
+
+        //2) Get the token by exchanging password
+        final TokenRequest ropcTokenRequest = oAuth2Strategy.createRopcTokenRequest(
+                parametersWithScopes
+        );
+
+        final TokenResult tokenResult = oAuth2Strategy.requestToken(ropcTokenRequest);
+
+        acquireTokenResult.setTokenResult(tokenResult);
+
+        @SuppressWarnings(WarningType.rawtype_warning) final OAuth2TokenCache tokenCache = parameters.getOAuth2TokenCache();
+
+        if (tokenResult != null && tokenResult.getSuccess()) {
+            // Suppressing unchecked warnings due to casting of rawtypes to generic types of OAuth2TokenCache's instance tokenCache while calling method saveAndLoadAggregatedAccountData
+            @SuppressWarnings(WarningType.unchecked_warning) final List<ICacheRecord> savedRecords = tokenCache.saveAndLoadAggregatedAccountData(
+                    oAuth2Strategy,
+                    getAuthorizationRequest(oAuth2Strategy, parameters),
+                    tokenResult.getTokenResponse()
+            );
+
+            final ICacheRecord savedRecord = savedRecords.get(0);
```

Comment:
check length first, or where will the NoSuchElementException go?

Reply:
probably caught by CommandDispatcher...same way it is happen in acquireTokenInteractive...it's the same code.